utils/cautils/certificate_flow.go

```diff
@@ -286,10 +286,25 @@ func (f *CertificateFlow) CreateSignRequest(ctx *cli.Context, tok, subject strin
 			dnsNames, ips, emails, uris = splitSANs(defaultSANs)
 		}
 	case token.OIDC:
-		if jwt.Payload.Email != "" {
-			emails = append(emails, jwt.Payload.Email)
+		// If no sans are given using the --san flag, and the subject argument
+		// matches the email then CN=token.sub SANs=email, token.iss#token.sub
+		//
+		// If no sans are given and the subject argument does not match the
+		// email then CN=subject SANs=splitSANs(subject)
+		//
+		// If sans are provided CN=subject SANs=splitSANs(sans)
+		if len(sans) == 0 {
```

Comment:
Do we need to be checking jwt.Payload.SANs here as well?

Reply:
I had it like that before, but I wanted to make it explicit when the `--san` flag was passed, and having a custom claim "sans" in an OIDC token would be quite weird.

Reply:
Fixed in 245ddf1